### DISCUSSION TITLE
"Unable to find plugin" error when deployed on Android

### DISCUSSION THREAD

**Discussion by grayac2** [ionic-team/capacitor#3743] [Q&A] [answered]
I'm working on an Android app that uses the Geolocation Capacitor plugin. It functioned perfectly on my test device when plugged up to Android Studio, but once I actually deploy the app bundle or APK to the device, I suddenly get the error "unable to find plugin," and the plugin doesn't load or function at all.
`Error: unable to find plugin : Geolocation
        at Object.fromNative (capacitor-runtime.js:230:14)
        at <anonymous>:1:18`

I've also just noticed this error that occurs when I open the app. Not sure if it is relevant.
```
TypeError: Cannot read property 'requestPermission' of undefined
        at t.initialize (http://localhost/main-es2015.b1e72c448947ebd7f44e.js:1:8205)
        at http://localhost/main-es2015.b1e72c448947ebd7f44e.js:1:1152476
        at a.invoke (http://localhost/polyfills-es2015.c1c77fc4a42126c82b7c.js:1:6465)
        at Object.onInvoke (http://localhost/main-es2015.b1e72c448947ebd7f44e.js:1:504506)
        at a.invoke (http://localhost/polyfills-es2015.c1c77fc4a42126c82b7c.js:1:6405)
        at s.run (http://localhost/polyfills-es2015.c1c77fc4a42126c82b7c.js:1:1974)
        at http://localhost/polyfills-es2015.c1c77fc4a42126c82b7c.js:1:12750
        at a.invokeTask (http://localhost/polyfills-es2015.c1c77fc4a42126c82b7c.js:1:7083)
        at Object.onInvokeTask (http://localhost/main-es2015.b1e72c448947ebd7f44e.js:1:504424)
        at a.invokeTask (http://localhost/polyfills-es2015.c1c77fc4a42126c82b7c.js:1:7004)
```

I have deployed the app through the Play store, as well as by manually installing the APK on my device. Both of these methods produced the same results. The location function still works when ran from Android Studio, however.

The android geolocation permission is configured.
```
<uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />

<uses-feature android:name="android.hardware.location.gps" />
```
The slider to activate location permissions is available in the device's settings, though an in-app prompt to allow it is never triggered since the plugin isn't working.

I can't find any information about this issue, so I've come here for help. If any more info is needed, I'll be happy to share anything that would be useful.

**Comment by jcesarmobile** [comment, marked as answer]
Have you changed the minifyEnabled value to true for release builds?
That prevents plugins from working

**Reply by grayac2** [reply]
I'm not sure how that got set, but that fixed the problem for me. Thank you for the help; I wish I would have come here sooner!

**Comment by Hamza01541** [comment]
Setting minifyEnabled to false in build.gradle file solves plugins not found issues:
`buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }`

**Reply by jcesarmobile** [reply]
Notice that this question is almost 3 years old, Capacitor was updated to not need to set minifyEnabled to false anymore.
If you are in Capacitor 5 and it's still not working, please, create an issue and provide a sample app so it can be fixed.
